state_transfer/transfer/rsync/options.go

```diff
@@ -0,0 +1,251 @@
+package rsync
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	labels "github.com/konveyor/crane-lib/state_transfer/labels"
+	transfer "github.com/konveyor/crane-lib/state_transfer/transfer"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	errorsutil "k8s.io/apimachinery/pkg/util/errors"
+)
+
+const (
+	optRecursive     = "--recursive"
+	optSymLinks      = "--links"
+	optPermissions   = "--perms"
+	optModTimes      = "--times"
+	optDeviceFiles   = "--devices"
+	optSpecialFiles  = "--specials"
+	optOwner         = "--owner"
+	optGroup         = "--group"
+	optHardLinks     = "--hard-links"
+	optPartial       = "--partial"
+	optDelete        = "--delete"
+	optBwLimit       = "--bwlimit=%d"
+	optInfo          = "--info=%s"
+	optHumanReadable = "--human-readable"
+	optLogFile       = "--log-file=%s"
+)
+
+const (
+	logFileStdOut = "/dev/stdout"
+)
+
+// TransferOptions defines customizeable options for Rsync Transfer
+type TransferOptions struct {
+	CommandOptions
+	sourceResourceMetadata      transfer.ResourceMetadata
+	destinationResourceMetadata transfer.ResourceMetadata
+}
+
+// TransferOption
+type TransferOption interface {
+	ApplyTo(*TransferOptions) error
+}
+
+func (rto *TransferOptions) Apply(opts ...TransferOption) (err error) {
```

Comment:
extreme NIT picking: change receiver name from `rto` to `t`, thats the conventions I have seen in most go code

```
func (m MyString) ...
```